### PR TITLE
[BM-584] Fix not saving notifications after reset

### PR DESCRIPTION
**Pull request by anyashka**
### Ticket

https://netguru.atlassian.net/browse/BM-584
 
### Task Description

 Fix not saving notifications after reset

### CODE REVIEW

File: Baby Monitor/Source Files/AppDelegate.swift [hunk 1]
```diff
@@ -13,12 +13,21 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
     var window: UIWindow?
     var rootCoordinator: RootCoordinatorProtocol?
-    let appDependencies = AppDependencies()
+    private(set) var appDependencies = AppDependencies()
     
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
         window = UIWindow(frame: UIScreen.main.bounds)
         rootCoordinator = RootCoordinator(window!, appDependencies: appDependencies)
         rootCoordinator?.start()
+        rootCoordinator?.onResetApp = { [weak self] in
```
Comment (pkozielecki) [inline Baby Monitor/Source Files/AppDelegate.swift]: I don't think this is needed. `ApplicationResetter` has `localResetCompletionObservable` we can subscribe to and get notified when app is done resetting.

File: Baby Monitor/Source Files/Modules/RootCoordinator.swift [hunk 2]
```diff
@@ -9,6 +9,7 @@ final class RootCoordinator: RootCoordinatorProtocol {
     
     var childCoordinators: [Coordinator] = []
     var onEnding: (() -> Void)?
+    var onResetApp: (() -> Void)?
```
Comment (pkozielecki) [inline Baby Monitor/Source Files/Modules/RootCoordinator.swift]: As a mentioned above, I think this method does not have to be declared.

File: Baby Monitor/Source Files/AppDelegate.swift [hunk 3]
```diff
@@ -13,12 +13,21 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+            // This line is extremely important. After resseting the app we may want to establish new
+            // WebRTC connection. Thanks to deinitializing and initializing AppDependencies again we are
+            // sure that old connection is properly cleared.
+            // UPD: TODO: Needed to check whether other view models are not keeping reference to the old dependencies.
+            let dependencies = AppDependencies()
+            self?.rootCoordinator?.update(dependencies: dependencies)
+            self?.appDependencies = dependencies
```
Comment (pkozielecki) [inline Baby Monitor/Source Files/AppDelegate.swift]: This is potentially very dangerous an leaky, but, for now, there is no alternative.

File: Baby Monitor/Source Files/AppDelegate.swift (hunk 1 above)
Comment (anyashka) [reply Baby Monitor/Source Files/AppDelegate.swift]: Totally! 👍

File: Baby Monitor/Source Files/AppDelegate.swift (hunk 3 above)
Comment (anyashka) [reply Baby Monitor/Source Files/AppDelegate.swift]: 😔

**Pull request merged**